apollo-router/src/plugins/connectors/handle_responses.rs

```diff
@@ -1428,6 +1428,10 @@ mod tests {
         ])
         .unwrap();
 
+        // Overwrite error IDs to avoid random Uuid mismatch
```

Comment:
Can you explain more about why this is needed?

Reply:
Fleshed out the comments.